### PR TITLE
Make rds files optional arguments. If absent, use reasonable defaults…

### PR DESCRIPTION
**Pull request by gfinak**
… from the faust package if these are not passed by the user.
The three files that are now optional are `channel_bounds.rds`, `active_channels.rds` and `supervised_list.rds`, consequently the parameters `--active_channels_path` , `--supervised_list_path`,  and `--channel_bounds_path` are optional and handled appropriately. They no longer need to be passed in. 
Any combination of the three config files can be given  (or not). 
FAUST has defaults for these that will be used if they are missing.

### CODE REVIEW

File: modules/processes/prepare_faust_data.nf [hunk 1]
```diff
@@ -46,11 +46,18 @@ process prepareFAUSTData {
             convert_legacy_gs("${input_gating_set_directory_channel}", converted_gating_set_directory_path)
             gating_set <- try(flowWorkspace::load_gs(converted_gating_set_directory_path))
         }
-
-        active_channels_rds_object <- readRDS("${active_channels_path_channel}")
-        channel_bounds_rds_object <- readRDS("${channel_bounds_path_channel}")
-        supervised_list_rds_object <- readRDS("${supervised_list_path_channel}")
-
+        active_channels_rds_object<-NULL
```
Comment (gfinak) [inline modules/processes/prepare_faust_data.nf:49]: The `null` test below requires these to be defined and set to null here.

File: modules/processes/prepare_faust_data.nf [hunk 2]
```diff
@@ -46,11 +46,18 @@ process prepareFAUSTData {
...
+        channel_bounds_rds_object<-NULL
+        supervised_list_rds_object<-NULL
+        if(file.exists("${active_channels_path_channel}")){
+            active_channels_rds_object <- readRDS("${active_channels_path_channel}")
+        }
+        if(file.exists("${channel_bounds_path_channel}")){
+            channel_bounds_rds_object <- readRDS("${channel_bounds_path_channel}")
+        }
+        if(file.exists("${supervised_list_path_channel}")){
```
Comment (gfinak) [inline modules/processes/prepare_faust_data.nf:58]: Test for file existence rather than testing for a null value.

File: main.nf [hunk 3]
```diff
@@ -32,11 +32,11 @@ if(total_parameters_provided_by_user == 0) {
...
 // FAUST Required
 params.input_gating_set_directory = "$baseDir/input"
-params.active_channels_path = "$baseDir/helper_files/active_channels.rds"
-params.channel_bounds_path = "$baseDir/helper_files/channel_bounds.rds"
-params.supervised_list_path = "$baseDir/helper_files/supervised_list.rds"
 
 // FAUST Optional
+params.channel_bounds_path = ""
+params.active_channels_path = ""
+params.supervised_list_path = ""
```
Comment (loganknecht) [inline main.nf:39]: Very cool. Those were legacy defaults! Thank you for changing this.

File: main.nf [hunk 4]
```diff
@@ -72,9 +72,22 @@ workflow {
...
+        }
+        if(params.channel_bounds_path){
+            channel_bounds_path = Channel.fromPath(params.channel_bounds_path)
+        }else{
+            channel_bounds_path = Channel.empty().ifEmpty("")
+        }
+        if(params.supervised_list_path){
+            supervised_list_path = Channel.fromPath(params.supervised_list_path)
+        }else{
+            supervised_list_path = Channel.empty().ifEmpty("")
```
Comment (loganknecht) [inline main.nf:89]: `Channel.empty().ifEmpty("")` - https://www.nextflow.io/docs/latest/operator.html#operator-ifempty

Super super super cool.

This is basically a tool to set default channels in the case of a missing channel. I like it!

File: modules/processes/prepare_faust_data.nf (hunk 1 above)
Comment (loganknecht) [reply modules/processes/prepare_faust_data.nf:49]: Good catch and great fix!

**Review by loganknecht — DISMISSED** [review]

**Review by loganknecht — APPROVED** [review]

**Comment by loganknecht** [comment]
@gfinak This was targeting the `master` branch (we should really rename it to `production` in light of things).

I changed it to target the `development` branch.

In order to get a merge from `development` into `master` we would have to cut a release

File: main.nf (hunk 3 above)
Comment (jacobpwagner) [reply main.nf:39]: I was just lamenting the loss of these defaults and making some dummy RDS files as I transition all of my first-pass HIPCCyto runs to FAUST_Nextflow. So, great timing on this.

**Pull request merged**